processor.go

```diff
@@ -442,6 +442,123 @@ func callInterfaceFunc(handler, input interface{}, streaming bool) (intf interfa
 	return
 }
 
+// ProcessClientStreamRequest allows clients to push multiple messages
+// asynchronously to the same service with the same connection. Unlike in
+// ProcessClientRequest, we take a channel of inputs that can be filled and will
+// subsequently call the service with any new messages received in the channel.
+func (p *ServiceProcessor) ProcessClientStreamRequest(req *http.Request, path string,
```

Comment:
Would it be possible to use this function inside `ProcessClientRequest` so that we reduce the code duplication ?

Reply:
I removed the handling of streaming transaction from ProcessClientRequest, which removes most of the duplication.